exec_test.go

```diff
@@ -494,3 +493,105 @@ func TestCommandCombinedOutput(t *testing.T) {
 		})
 	}
 }
+
+func TestCommandStdin(t *testing.T) {
+	d := newTestDistro(t, jammyRootFs)
+
+	const (
+		readFromPipe int = iota + 1
+		readFromBuffer
+	)
+
+	testCases := map[string]struct {
+		text string // We'll write this to Stdin. A python program will read it back to us.
+		// It is therefore both input and part of the expected output.
+
+		closeBeforeWait bool // Set to true to close the pipe before execution of the Cmd is over
+		readFrom        int  // Where Cmd should read Stdin from
+	}{
+		"standard":         {text: "Hello, wsl!", readFrom: readFromPipe},
```

Comment:
Ditto here, the text is the same for 3 over the 4 tests. We could have in the test itself:
```go
if tc.text == "" {
   tc.text = "Hello, wsl!"
}
```

in the setup part of the subtest.

Then, with this change and the previous suggested one, each test case definition become very clear on their variation in the test fixtures:
```go
		"standard":         {},
		"funny characters": {text: "Hello, \x00\twsl!"},
		"closing early":    {closeBeforeWait: true},
		"using a buffer":   {readFrom: readFromBuffer},
```

Reply:
That makes a lot of sense, done.